lib/exchange/provider/trocador_exchange_provider.dart

```diff
@@ -153,6 +170,22 @@ class TrocadorExchangeProvider extends ExchangeProvider {
       params['id'] = _lastUsedRateId;
     }
 
+
+    String firstAvailableProvider = '';
+
+      for (var provider in _provider) {
+        if (providerStates.containsKey(provider) && providerStates[provider] == true) {
+          firstAvailableProvider = provider as String;
+          break;
+        }
+      }
+
+      if (firstAvailableProvider.isEmpty) {
+        throw Exception('No available provider is enabled');
+      }
+
+      params['provider'] = firstAvailableProvider;
```

Comment:
please fix the indentation.

and are we sure that `firstAvailableProvider` would have the best rate? or should we loop through the enabled providers and their rates and choose depending on the best rate?

Reply:
The list of available providers for a trade is sorted according to the 'amount_to' parameter, placing the highest amounts at the top. Therefore, it's already sorted to highlight the best conditions. Since we fetch the first allowed provider from this list, it seems we are indeed getting the best rate.